src/builder/QuarkBuilder.sol

```diff
@@ -51,6 +51,170 @@ contract QuarkBuilder {
+            // but if you're borrowing the payment token, you can use the
+            // borrowed amount to cover the cost
+
+            if (Strings.stringEqIgnoreCase(payment.currency, borrowIntent.assetSymbol)) {
+                maxCostOnDstChain = Math.subtractFlooredAtZero(maxCostOnDstChain, borrowIntent.amount);
+            }
+
+            if (
+                needsBridgedFunds(payment.currency, maxCostOnDstChain, borrowIntent.chainId, chainAccountsList, payment)
+            ) {
+                (IQuarkWallet.QuarkOperation[] memory bridgeQuarkOperations, Actions.Action[] memory bridgeActions) =
+                Actions.constructBridgeOperations(
+                    Actions.BridgeOperationInfo({
+                        assetSymbol: payment.currency,
+                        amountNeededOnDst: maxCostOnDstChain,
+                        dstChainId: borrowIntent.chainId,
+                        recipient: borrowIntent.borrower,
+                        blockTimestamp: borrowIntent.blockTimestamp,
+                        useQuotecall: false // XXX support Quotecall
+                    }),
+                    chainAccountsList,
+                    payment
+                );
+
+                for (uint256 i = 0; i < bridgeQuarkOperations.length; ++i) {
+                    quarkOperations[actionIndex] = bridgeQuarkOperations[i];
+                    actions[actionIndex] = bridgeActions[i];
+                    actionIndex++;
+                }
+            }
+        }
+
+        (quarkOperations[actionIndex], actions[actionIndex]) = Actions.borrow(
+            Actions.BorrowInput({
+                chainAccountsList: chainAccountsList,
+                amount: borrowIntent.amount,
+                assetSymbol: borrowIntent.assetSymbol,
+                blockTimestamp: borrowIntent.blockTimestamp,
+                borrower: borrowIntent.borrower,
+                chainId: borrowIntent.chainId,
+                collateralAmounts: borrowIntent.collateralAmounts,
+                collateralTokenSymbols: borrowIntent.collateralTokenSymbols,
+                comet: borrowIntent.comet
+            }),
+            payment
+        );
+
+        actionIndex++;
+
+        // Truncate actions and quark operations
+        actions = Actions.truncate(actions, actionIndex);
+        quarkOperations = Actions.truncate(quarkOperations, actionIndex);
+
+        // Validate generated actions for affordability
+        if (payment.isToken) {
+            uint256 supplementalPaymentTokenBalance = 0;
+            if (Strings.stringEqIgnoreCase(payment.currency, borrowIntent.assetSymbol)) {
+                supplementalPaymentTokenBalance += borrowIntent.amount;
+            }
+
+            assertSufficientPaymentTokenBalances(
+                actions, chainAccountsList, borrowIntent.chainId, supplementalPaymentTokenBalance
+            );
+        }
+
```

Comment:
You'll want to merge the operations and optionally wrap around paycall/quotecall here. See https://github.com/compound-finance/quark-scripts/blob/0cf5cc6ee5649a4b72204912559be01e8dee5e9d/src/builder/QuarkBuilder.sol#L151-L160

Reply:
updated